File: 1.6/1.6.js
var compressString = function(str) {
  var a = str.split('');
  var output = '';
  var i = 0;
  while (i < a.length) {
    var count = 1;
    var j = 1;
    while (a[i] === a[i + j]) {
      count += 1;
      j++;
    }
    output += (a[i] + count.toString());
    i += count;
  };
  return output;
};

x = 'aaabbbccc'
y = 'aaaabcc'

File: 1.6/1.6.rb
def compress_string(str)
  chars = str.chars
  output = ''
  chars.each_with_index do |c,i|
    count = 1
    addn = ''
    if c == chars[i-1]
      count += 1
    else
      addn += c
    end
    output += c
  end
end